pkg/deploy/lattice/access_log_subscription_manager.go

```diff
@@ -126,6 +127,47 @@ func (m *defaultAccessLogSubscriptionManager) Create(
 	}, nil
 }
 
+func (m *defaultAccessLogSubscriptionManager) Update(
+	ctx context.Context,
+	accessLogSubscription *lattice.AccessLogSubscription,
+) (*lattice.AccessLogSubscriptionStatus, error) {
+	vpcLatticeSess := m.cloud.Lattice()
+	updateALSInput := &vpclattice.UpdateAccessLogSubscriptionInput{
+		AccessLogSubscriptionIdentifier: aws.String(accessLogSubscription.Status.Arn),
+		DestinationArn:                  aws.String(accessLogSubscription.Spec.DestinationArn),
+	}
+	updateALSOutput, err := vpcLatticeSess.UpdateAccessLogSubscriptionWithContext(ctx, updateALSInput)
+	if err != nil {
+		switch e := err.(type) {
+		case *vpclattice.AccessDeniedException:
+			return nil, services.NewInvalidError(e.Message())
+		case *vpclattice.ResourceNotFoundException:
+			if *e.ResourceType == "SERVICE_NETWORK" || *e.ResourceType == "SERVICE" {
+				return nil, services.NewNotFoundError(string(accessLogSubscription.Spec.SourceType), accessLogSubscription.Spec.SourceName)
+			}
+			return nil, services.NewInvalidError(e.Message())
+		case *vpclattice.ConflictException:
+			/*
+			 * A conflict can happen when the destination type of the new ALS is different from the original.
+			 * To gracefully handle this, we create a new ALS with the new destination, then delete the old one.
+			 */
+			alsStatus, err := m.Create(ctx, accessLogSubscription)
+			if err != nil {
+				return nil, err
+			}
+			err = m.Delete(ctx, accessLogSubscription)
+			if err != nil {
+				return nil, err
+			}
+			return alsStatus, nil
+		}
+	}
+
+	return &lattice.AccessLogSubscriptionStatus{
+		Arn: *updateALSOutput.Arn,
+	}, nil
```

Comment:
Since you are getting into de-nesting code. You can change this one too.

```go
if err == nil {
     return &lattice.AccessLogSubscriptionStatus{
         Arn: *updateALSOutput.Arn,
     }, nil
}
swtich ...
```

Reply:
Good point, will do